pkg/agent/controller/bgp/controller.go

```diff
@@ -0,0 +1,954 @@
+		return
+	}
+	if c.hasAffectedBPsByService(svc) {
+		klog.V(2).InfoS("Processing EndpointSlice ADD event", "EndpointSlice", klog.KObj(eps))
+		c.queue.Add(key)
+	}
+}
+
+func (c *Controller) updateEndpointSlice(oldObj, obj interface{}) {
+	eps := obj.(*discovery.EndpointSlice)
+	svc, _ := c.serviceLister.Services(eps.GetNamespace()).Get(eps.GetLabels()[discovery.LabelServiceName])
+	if svc == nil {
+		return
+	}
+	if svc.Spec.ExternalTrafficPolicy == corev1.ServiceExternalTrafficPolicyTypeCluster &&
+		(svc.Spec.InternalTrafficPolicy == nil || *svc.Spec.InternalTrafficPolicy == corev1.ServiceInternalTrafficPolicyCluster) {
+		return
+	}
+	affectedBPs := c.filterAffectedBPsByService(svc)
+	if len(affectedBPs) != 0 {
+		klog.V(2).InfoS("Processing EndpointSlice UPDATE event", "EndpointSlice", klog.KObj(eps))
+		c.queue.Add(key)
+	}
+}
+
+func (c *Controller) deleteService(obj interface{}) {
+	svc := obj.(*corev1.Service)
+	affectedBPs := c.filterAffectedBPsByService(svc)
+	if len(affectedBPs) != 0 {
+		klog.V(2).InfoS("Processing Service DELETE event", "Service", klog.KObj(svc))
+		c.queue.Add(key)
+	}
+}
+
+func (c *Controller) hasAffectedBPsByEgress() bool {
+	allBPs, _ := c.bgpPolicyLister.List(labels.Everything())
+	for _, bp := range allBPs {
+		if !c.matchedCurrentNode(bp) {
+			continue
+		}
+		if bp.Spec.Advertisements.Egress != nil {
+			return true
+		}
+	}
+	return false
+}
+
+func (c *Controller) addEgress(obj interface{}) {
+	if !c.egressEnabled {
+		return
+	}
+	eg := obj.(*v1beta1.Egress)
+	if eg.Status.EgressNode != c.nodeName {
+		return
+	}
+	if c.hasAffectedBPsByEgress() {
+		klog.V(2).InfoS("Processing Egress ADD event", "Egress", klog.KObj(eg))
+		c.queue.Add(key)
+	}
+}
+
+func (c *Controller) updateEgress(oldObj, obj interface{}) {
+	if !c.egressEnabled {
+		return
+	}
+	oldEg := oldObj.(*v1beta1.Egress)
+	eg := obj.(*v1beta1.Egress)
+	if oldEg.Status.EgressNode != c.nodeName && eg.Status.EgressNode != c.nodeName {
+		return
+	}
+	if oldEg.Status.EgressIP == eg.Status.EgressIP {
+		return
+	}
+	if c.hasAffectedBPsByEgress() {
+		klog.V(2).InfoS("Processing Egress UPDATE event", "Egress", klog.KObj(eg))
+		c.queue.Add(key)
+	}
+}
+
+func (c *Controller) deleteEgress(obj interface{}) {
+	if !c.egressEnabled {
+		return
+	}
+	eg := obj.(*v1beta1.Egress)
+	if eg.Status.EgressNode != c.nodeName {
+		return
+	}
+	if c.hasAffectedBPsByEgress() {
+		klog.V(2).InfoS("Processing Egress DELETE event", "Service", klog.KObj(eg))
+		c.queue.Add(key)
+	}
+}
+
+func (c *Controller) filterAffectedBPsByNode(node *corev1.Node) sets.Set[string] {
+	affectedBPs := sets.New[string]()
+	allBPs, _ := c.bgpPolicyLister.List(labels.Everything())
+	for _, bp := range allBPs {
+		if c.matchedNode(node, bp) {
+			affectedBPs.Insert(bp.GetName())
+		}
+	}
+	return affectedBPs
+}
+
+func (c *Controller) updateNode(oldObj, obj interface{}) {
+	oldNode := oldObj.(*corev1.Node)
+	node := obj.(*corev1.Node)
```

Comment:
Name this to newNode to keep name format consistent.

Reply:
I saw the code of other existing coontroller is like this: `old<object name>` and just `<object name>` without `new`.